_posts/2021-07-30-release-0.9.12.md

```diff
@@ -0,0 +1,314 @@
+
+## Optical Flow Camera
+
+The Optical Flow camera is a new sensor that captures the motion perceived from the point of view of the camera. Every pixel recorded by the Optical Flow camera encodes the velocity of that point projected to the image plane. You can find the sensor reference material and additional attributes [here](https://carla.readthedocs.io/en/latest/ref_sensors/#optical-flow-camera).
+
+![optical_flow](/img/posts/2021-07-30/optical_flow.png){:class="img-fluid rounded mx-auto d-block"}
+
+## New and Improved Vehicles
+
+The CARLA garage has been busy with five new vehicles added to the CARLA blueprint library!
+
+![vehicles](/img/posts/2021-07-30/vehicles.png){:class="img-fluid rounded mx-auto d-block"}
+
+Introducing from left to right:
+
+- Firetruck
+- Mini Cooper 2021
+- Nissan Patrol 2021
+- Ambulance
+- Mercedes Sprinter
+
+As well as completely new vehicles, we have generally improved the vehicle materials and meshes. Dirt has been added to vehicle exteriors to increase realism. Mesh geometry has been increased to improve appearance.
+
+Finally, some of our vehicle names have changed in the blueprint library. These changes include:
+
+- `vehicle.bmw.isetta -> vehicle.micro.microlino`
+- `vehicle.dodge_charger.police -> vehicle.dodge.charger_police`
+- `vehicle.mercedes-benz.coupe -> vehicle.mercedes.coupe`
+- `vehicle.mini.cooperst -> vehicle.mini.cooper_s`
+- `vehicle.lincoln.mkz2017 -> vehicle.lincoln.mkz_2017`
+- `vehicle.mustang.mustang -> vehicle.ford.mustang`
+- `vehicle.lincoln2020.mkz2020 -> vehicle.lincoln.mkz_2020`
+- `vehicle.mercedesccc.mercedesccc -> vehicle.mercedes.coupe_2020`
+- `vehicle.charger2020.charger2020 -> vehicle.dodge.charger_2020`
+- `vehicle.chargercop2020.chargercop2020 -> vehicle.dodge.charger_police_2020`
+- `vehicle.vespa.vespa -> vehicle.vespa.zx125`
+
+## New and Improved Pedestrians
+
+We welcome a diverse new set of pedestrians to the CARLA family. A new generation of pedestrians has been added to the [blueprint library](https://carla.readthedocs.io/en/latest/bp_library/). The new pedestrians have a variety of body shapes and skin colors. Pedestrians all over the board have been improved with new shaders for skin, eyes, and clothing that add microdetail information. Meshes have been improved with a higher polygon count and better topology.
+
+![pedestrians](/img/posts/2021-07-30/pedestrians.png){:class="img-fluid rounded mx-auto d-block"}
+
+Actors in the blueprint library now have an additional generation attribute to distinguish between old and new models when you spawn them. You can also set some flags in `generate_traffic.py` and `manual_control.py` that defines which generations you would like to spawn:
+
+```sh
+# In generate_traffic.py
+# Generate first generation vehicles
+--generationv "1"
```

Comment:
We don't need the quotes with the parameter. It should be 
--generationv 1
--generationv 2
--generationv All